controllers/cloudstackfailuredomain_controller.go

```diff
@@ -107,6 +114,43 @@ func (r *CloudStackFailureDomainReconciliationRunner) Reconcile() (retRes ctrl.R
 	return ctrl.Result{}, nil
 }
 
+func (r *CloudStackFailureDomainReconciliationRunner) SetOwnerReferenceToEndpointSecret() error {
+	secret, err := r.GetSecretForFailureDomain(&r.ReconciliationSubject.Spec)
+	if err != nil {
+		return err
+	}
+	if MaySetOwnerReference(secret, r.CAPICluster) {
+		return r.Patcher.Patch(r.RequestCtx, secret)
+	}
+	return nil
+}
+
+func MaySetOwnerReference(secret *corev1.Secret, cluster *clusterv1.Cluster) bool {
```

Comment:
The name of this function makes me think it's const (i.e., returning a bool indicating whether you may set the owner reference).  Since it's not, I'd prefer a name that indicates so, maybe OptionallySetOwnerReference.